tests/test_device.py

```diff
@@ -20,6 +20,38 @@
 
 from conftest import shortnames
 from pennylane_ionq.api_client import ResourceManager, Job, Field
+from pennylane_ionq.device import QPUDevice
+
+FAKE_API_KEY = "ABC123"
+
+
+class TestDevice:
+    """Tests for the IonQDevice class."""
+
+    @pytest.mark.parametrize(
+        "wires,histogram",
+        [
+            (1, {"0": 0.6, "1": 0.4}),
+            (2, {"0": 0.1, "1": 0.2, "2": 0.3, "3": 0.4}),
+            (4, {"0": 0.413, "6": 0.111, "15": 0.476}),
+            (4, {"7": 0.413, "3": 0.111, "2": 0.476}),
+        ],
+    )
+    def test_generate_samples_qpu_device(self, wires, histogram):
+        """Test that the generate_samples method for QPUDevices shuffles the samples between calls."""
+
+        dev = QPUDevice(wires, shots=1024, api_key=FAKE_API_KEY)
+        dev.histogram = histogram
+
+        sample1 = dev.generate_samples()
+        sample2 = dev.generate_samples()
+        assert not np.all(sample1 == sample2)  # some rows are different
```

Comment:
(Minor): Maybe it can ease readability to have an extra line in between asserts.
```suggestion
        assert not np.all(sample1 == sample2)  # some rows are different

```